diff --git a/node_modules/@mantine/core/esm/core/Box/use-random-classname/use-random-classname.mjs b/node_modules/@mantine/core/esm/core/Box/use-random-classname/use-random-classname.mjs
index 494921c..041fcff 100644
--- a/node_modules/@mantine/core/esm/core/Box/use-random-classname/use-random-classname.mjs
+++ b/node_modules/@mantine/core/esm/core/Box/use-random-classname/use-random-classname.mjs
@@ -1,8 +1,40 @@
 'use client';
-import { useId } from 'react';
+import React from 'react';
+
+/**
+ * Credit: https://github.com/reach/reach-ui/blob/dev/packages/auto-id/src/auto-id.ts
+ */
+let handoffComplete = false;
+let id = 0;
+const genId = () => ++id;
+const useId = (prefix, idProp) => {
+  const initialId = idProp || (handoffComplete ? genId() : null);
+  const [uid, setUid] = React.useState(initialId);
+
+  React.useLayoutEffect(() => {
+    if (uid === null) setUid(genId());
+  }, []);
+
+  React.useEffect(() => {
+    if (!handoffComplete) {
+      handoffComplete = true;
+    }
+  }, []);
+
+  const id = uid != null ? uid.toString() : undefined;
+  return prefix ? `${prefix}-${id}` : id;
+};
 
 function useRandomClassName() {
-  const id = useId().replace(/:/g, "");
+  let id = '';
+
+  const isReact18 = React.useId !== undefined;
+  if (isReact18) {
+    id = React.useId().replace(/:/g, "");
+  } else {
+    id = useId('cls').replace(/:/g, "");
+  }
+
   return `__m__-${id}`;
 }
 
