diff --git a/node_modules/react-native-reanimated/lib/reanimated2/jestUtils.js b/node_modules/react-native-reanimated/lib/reanimated2/jestUtils.js
index d8befd8..00c1752 100644
--- a/node_modules/react-native-reanimated/lib/reanimated2/jestUtils.js
+++ b/node_modules/react-native-reanimated/lib/reanimated2/jestUtils.js
@@ -145,14 +145,21 @@ export const advanceAnimationByFrame = (count) => {
     jest.advanceTimersByTime(frameTime);
 };
 export const setUpTests = (userConfig = {}) => {
-    let expect;
-    try {
-        expect = require('expect');
-    }
-    catch (_) {
-        // for Jest in version 28+
-        const { expect: expectModule } = require('@jest/globals');
-        expect = expectModule;
+    let expect = global.expect;
+    if (expect === undefined) {
+      const expectModule = require('expect');
+      expect = expectModule;
+      // Starting from Jest 28, "expect" package uses named exports instead of default export.
+      // So, requiring "expect" package doesn't give direct access to "expect" function anymore.
+      // It gives access to the module object instead.
+      // We use this info to detect if the project uses Jest 28 or higher.
+      if (typeof expect === 'object') {
+        const jestGlobals = require('@jest/globals');
+        expect = jestGlobals.expect;
+      }
+      if (expect === undefined || expect.extend === undefined) {
+        expect = expectModule.default;
+      }
     }
     require('setimmediate');
     frameTime = Math.round(1000 / config.fps);
